apps/server/src/controllers/ontimeController.validate.ts

```diff
@@ -166,3 +170,101 @@ export const validateLoadProjectFile = [
     next();
   },
 ];
+
+/**
+ * @description Validates the filenames for duplicating a project.
+ */
+export const validateProjectDuplicate = [
+  body('newFilename')
+    .exists()
+    .withMessage('New project filename is required')
+    .isString()
+    .isLength({ min: 1, max: 255 })
+    .withMessage('New project filename must be a string'),
+
+  (req, res, next) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    next();
+  },
+];
+
+/**
+ * @description Validates the filenames for renaming a project.
+ */
+export const validateProjectRename = [
+  body('newFilename')
+    .exists()
+    .withMessage('Duplicate project filename is required')
+    .isString()
+    .isLength({ min: 1, max: 255 })
+    .withMessage('Duplicate project filename must be a string'),
+
+  (req, res, next) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    next();
+  },
+];
+
+/**
+ * @description Validates the filename for creating a project file.
+ */
+export const validateProjectCreate = [
+  body('filename')
+    .exists()
+    .withMessage('Filename is required')
+    .isString()
+    .isLength({ min: 1, max: 255 })
+    .withMessage('Filename must be a string'),
+  (req, res, next) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+
+    next();
+  },
+];
+
+const checkExistingProjectFile = (projectFilename?: string): boolean => {
+  const projectFilePath = join(uploadsFolderPath, projectFilename);
+
+  return existsSync(projectFilePath);
+};
```

Comment:
question: I am not entirely sure this function needs to exist, what do you think?

Reply:
you're right, it doesn't need to be its own function now that we reduced the logic for that check